### PR TITLE
Use S3 client instead of Minio client

### PR DESCRIPTION
**Pull request by zachstence**
Swap the Minio client for the basic AWS S3 client so we can connect to any S3 compliant backend

I have tested replay parse pipeline locally
I have tested image gen endpoints in isolation (need some more help validating that we're good there)
I also removed some (seemingly) unused endpoints from image gen

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,5 +45,8 @@
     "*.ts": "eslint --fix",
     "*.svelte": "eslint --fix",
     "*.js": "eslint --fix"
+  },
+  "dependencies": {
```
Comment (bwpikaard) [inline package.json:49]: Wrong package?

File: clients/discord-bot/config/default.json [hunk 2]
```diff
@@ -14,15 +14,15 @@
   "gql": {
     "url": "http://localhost:3001/graphql"
   },
-  "logger": {
-    "levels": true
-  },
-  "minio": {
-    "endPoint": "",
+  "s3": {
```
Comment (bwpikaard) [inline clients/discord-bot/config/default.json:17]: Do we want these in here?

**Review by bwpikaard — CHANGES_REQUESTED** [review]

File: clients/image-generation-frontend/src/config.template.json [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
-        "user": "",
-        "database": "",
-        "enable_logs": false,
-        "ssl":{ "rejectUnauthorized": false }
-      },
-    "transport": {
-        "url": "",
-        "queue": "dev-image-generation"
+  "s3": {
+    "endpoint": "https://s3.us-west-004.backblazeb2.com",
```
Comment (gankoji) [inline clients/image-generation-frontend/src/config.template.json]: config value in template

File: package.json (hunk 1 above)
Comment (zachstence) [reply package.json]: [687e0a2](https://github.com/SprocketBot/sprocket/pull/457/commits/687e0a2a6f9d28a80568b45437ed9cc84d2b5949)

File: clients/image-generation-frontend/src/routes/api/exec/[id].ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
-// import type {EndpointOutput, Request} from "@sveltejs/kit";
+import type {RequestHandler} from "@sveltejs/kit";
```
Comment (gankoji) [inline clients/image-generation-frontend/src/routes/api/exec/[id].ts:1]: Not splitting a svelte refactor from your minio to S3 change, rookie move XD

File: clients/discord-bot/config/default.json (hunk 2 above)
Comment (zachstence) [reply clients/discord-bot/config/default.json:17]: [b77d469](https://github.com/SprocketBot/sprocket/pull/457/commits/b77d4698364dc08bc2ce5204d8394fe0b0b33e3a)

File: microservices/image-generation-service/package.json [hunk 5]
```diff
@@ -30,7 +30,6 @@
     "@nestjs/microservices": "^8.2.3",
     "@types/jsdom": "^16.2.13",
     "@types/knex": "^0.16.1",
-    "@types/minio": "^7.0.10",
```
Comment (gankoji) [inline microservices/image-generation-service/package.json:33]: This seems like it ought to now depend on the AWS client, but maybe I'm crazy

**Review by bwpikaard — APPROVED** [review]

**Review by gankoji — APPROVED** [review]

**Pull request merged**